lib/addons/core-addon/index.ts

```diff
@@ -40,30 +45,43 @@ export class CoreAddOn implements ClusterAddOn {
 
     deploy(clusterInfo: ClusterInfo): Promise<Construct> {
 
-        // Create a service account if user provides namespace and service account
+        
         let serviceAccountRoleArn: string | undefined = undefined;
+        let addonVersion: string | undefined = undefined;
         let serviceAccount: ServiceAccount | undefined = undefined;
+        let saNamespace: string | undefined = undefined;
 
+        saNamespace = DEFAULT_NAMESPACE;
+        if (this.coreAddOnProps?.namespace) {
+            saNamespace = this.coreAddOnProps.namespace;
+        }
+
+        // Create a service account if user provides namespace, PolicyDocument
         if (this.coreAddOnProps?.policyDocumentProvider) {
             const policyDoc = this.coreAddOnProps.policyDocumentProvider(clusterInfo.cluster.stack.partition);
             serviceAccount  = createServiceAccount(clusterInfo.cluster, this.coreAddOnProps.saName,
-                DEFAULT_NAMESPACE, policyDoc);
+                saNamespace, policyDoc);
             serviceAccountRoleArn = serviceAccount.role.roleArn;
-
         }
-        const cfnaddon = new CfnAddon(clusterInfo.cluster.stack, this.coreAddOnProps.addOnName + "-addOn", {
+
+        let addOnprops = {
```

Comment:
minor: let's use camel case consistently - `addOnProps`

Reply:
Fixed all in CoreAddon PR.